### ISSUE TITLE
Pretty JSON doesn't output numeric literals properly which exceed 15 characters

### ISSUE THREAD

**Issue by carymrobbins** [a85/POSTMan-Chrome-Extension#2703]
1. Postman Version: Version 4.10.0
 2. App (Chrome app or Mac app): Chrome and Mac
 3. OS details: Reproduced consistently on Mac and Linux
 4. Is the Interceptor on and enabled in the app: No
 5. Did you encounter this recently, or has this bug always been there: Recently
 6. Expected behaviour: Output numeric literals in their original form.

## Steps to Reproduce

From a terminal, create a dummy JSON file which contains numeric literals which exceed 15 characters.

```
% echo '{"a":1234567890123456,"b":12345.6789012345}' > foo.json
% python2 -m SimpleHTTPServer
Serving HTTP on 0.0.0.0 port 8000 ...
```

Go to Postman and issue a `GET localhost:8000/foo.json`. The **Raw** output displays the original file text. However, the **Pretty JSON** output displays some weird formatting, possibly some sort of scientific notation. Here is the erroneous output -

```json
{
  "a": {
    "s": 1,
    "e": 15,
    "c": [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      0,
      1,
      2,
      3,
      4,
      5,
      6
    ]
  },
  "b": {
    "s": 1,
    "e": 4,
    "c": [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      0,
      1,
      2,
      3,
      4,
      5
    ]
  }
}
```

**Comment by SamvelRaja** [comment]
@carymrobbins Thanks for bringing this to our notice, its been identified and will update the thread once it is released 😄

**Comment by vsethi13** [comment]
I faced the same issue. Spent good 30 minutes debugging my code just to realize the problem is with rendering in Postman. Reverted back to postman v4.9.3

**Comment by adam-sanchez** [comment]
same issues this morning so have had to revert back to Version 4.9.3 in the meantime yes

**Comment by reger** [comment]
Having same problem with v4.10.0 on mac, and like @vsethi13 wasted a bunch of time trying to track down a code issue before realizing it was Postman... Will be reverting back to an earlier version as well.

**Comment by zmorris** [comment]
Had this same issue, when I went to `Postman->Code->Shell cURL` and ran the request from console, the output was correct.  Thankfully a coworker had me try Raw instead of Pretty or I would have been here all day thinking it was some subtle debug feature in the backend codebase.  For now I am just pasting the output into JSON Editor to see it pretty printed:

Online:
http://www.jsoneditoronline.org/

Extension:
https://chrome.google.com/webstore/detail/json-editor/lhkmoheomjbkfloacpgllgjcamhihfaj?hl=en

**Comment by ben-reilly** [comment]
Wish someone had suggested the same thing to me: I _did_ spend all day thinking it was something in our backend code. 😄

I hope someone creates a test for this.

**Comment by SamvelRaja** [comment]
Here all, the fix has been released with native mac and Linux 4.10.1. Native windows and chrome will be updated with the fix soon

**Comment by robertkuhar** [comment]
This is definitely NOT FREAKING FIXED on OS X!??!?!?!  @SamvelRaja The churn on Postman is becoming untenable for me.  How do I downgrade Postman?  This is starting to feel like "amateur hour" to me.

This fails always for me with actual backend services that are emitting proper JSON at version Version 4.10.1 (4.10.1)

**Comment by hn3000** [comment]
While @robertkuhar is the best kind of correct (technically correct), the situation has improved. Instead of a hot garbled mess I now see strings instead of numbers.

Actual JSON:
<img width="563" alt="screen shot 2017-02-15 at 20 01 16" src="https://cloud.githubusercontent.com/assets/133752/22990494/adee0640-f3b9-11e6-9a5c-dd3275d53a06.png">

Pretty-Printed JSON:
<img width="416" alt="screen shot 2017-02-15 at 20 01 09" src="https://cloud.githubusercontent.com/assets/133752/22990516/bdc74ad6-f3b9-11e6-887b-cd5b57d8aee3.png">

While some may find strings prettier than numbers, I would rather see what the server sent.

**Comment by sdnts** [comment]
@robertkuhar Can you upload a screenshot of what you see (with sensitive data redacted) here?

**Comment by robertkuhar** [comment]
@madebysid 

Pretty is WRONG!  ![pretty_is_wrong](https://cloud.githubusercontent.com/assets/190304/22992085/702c8c14-f373-11e6-82f6-0b775be56cf8.png)
Raw is Correct.  ![raw_is_correct](https://cloud.githubusercontent.com/assets/190304/22992092/79989bda-f373-11e6-8c7a-865c722a971b.png)
Standalone cURL agrees.  ![standalone_curl_agrees_with_raw](https://cloud.githubusercontent.com/assets/190304/22992159/a2a2b95c-f373-11e6-8b4f-57ede4cf32fe.png)

About says I am Version 4.10.1 (4.10.1)

**Comment by robertkuhar** [comment]
The problem is not just arrays.  Numbers at the top level are incorrectly changed to strings on the Pretty display.   The latitude and longitude emitted on these screenshots came out of a java service as doubles.

![pretty_is_wrong 2](https://cloud.githubusercontent.com/assets/190304/22993097/9e410f64-f376-11e6-9d50-824033871663.png)
![raw_is_correct 2](https://cloud.githubusercontent.com/assets/190304/22993098/9e418eb2-f376-11e6-8a33-f93ddf201af8.png)
![standalone_curl_agrees 2](https://cloud.githubusercontent.com/assets/190304/22993099/9e4c13fa-f376-11e6-858c-17cf7af23b8b.png)

**Comment by SamvelRaja** [comment]
@robertkuhar @hn3000  thanks for pointing me to this edge case which came out while solving this issue. We had updated the fix and released in 4.10.2

**Comment by robertkuhar** [comment]
@SamvelRaja I do not share the opinion that this is an "edge case".  Failure to correctly render the JSON sent by an the endpoint is a really big deal.  I really do remain shocked that there doesn't appear to be adequate testing of these code paths that could have detected this discrepancy.  I hope there is now.  A Number in one view must always be rendered as a Number in another.

My faith in Postman has slid backwards the past couple of 3 releases.  I am hopeful that the quality will improve as this had been a very valuable tool for allowing me to drive our micro-services by hand.
